src/libdredd/src/mutate_visitor.cc

```diff
@@ -108,14 +108,23 @@ bool MutateVisitor::TraverseDecl(clang::Decl* decl) {
     // consider it for mutation.
     return true;
   }
-  if (llvm::dyn_cast<clang::StaticAssertDecl>(decl) != nullptr) {
+  if (const auto* static_assert_decl =
+          llvm::dyn_cast<clang::StaticAssertDecl>(decl)) {
     // It does not make sense to mutate static assertions, as (a) this will
     // very likely lead to compile-time failures due to the assertion not
     // holding, (b) if compilation succeeds then the assertion is not actually
     // present at runtime so there is no notion of killing the mutant, and (c)
     // dynamic instrumentation of the mutation operator will break the rules
     // associated with static assertions anyway.
-    return true;
+
+    // However, some C++ compiler complain if the expression under static
+    // assertion is not a compile-time constants. To allow the constant
+    // declaration that occurs inside static assertion to have their initial
+    // value mutated (because they may be used elsewhere), but to avoid
+    // compilaton errors, we record the static assertion so that later their
+    // argument expression can be replaced with the value to which the argument
+    // expression would notmally evaluate.
```

Comment:
notmally -> normally. However, I think you noted in a comment on the issue that actually this gets replaced with 1 later, so can you update this comment.